src/farbfeld.rs

```diff
@@ -101,6 +91,76 @@ impl<R: Read> Read for FarbfeldReader<R> {
     }
 }
 
+impl<R: Read + Seek> Seek for FarbfeldReader<R> {
+    fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
+        fn checked_seek<F: FnOnce(u64) -> io::Result<()>>(from: u64, by: i64, cbk: F) -> io::Result<bool> {
+            match if by < 0 {
+                from.checked_sub(by.wrapping_neg() as u64)
+            } else {
+                from.checked_add(by as u64)
+            } {
+                Some(off) if off < ORIGIN => Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid seek to a negative position")),
+                Some(off) => {
+                    cbk(off)?;
+                    Ok(off % 2 == 1)
```

Comment:
This will allow seeking beyond the end, and then read a byte into the buffer. (If `SeekFrom::End` with a positive but odd offset is called).

Reply:
Done